packages/editor/src/components/start-page-options/index.js

```diff
@@ -27,24 +22,34 @@ export default function StartPageOptions() {
 		);
 		return {
 			postId: getCurrentPostId(),
-			shouldEnable:
+			enabled:
 				choosePatternModalEnabled &&
 				! preferencesModalActive &&
-				! isEditedPostDirty() &&
-				isEditedPostEmpty() &&
 				'page' === getCurrentPostType(),
 		};
 	}, [] );
+	const { isEditedPostDirty, isEditedPostEmpty } = useSelect( editorStore );
 	const { setIsInserterOpened } = useDispatch( editorStore );
 
 	useEffect( () => {
-		if ( shouldEnable ) {
+		if ( ! enabled ) {
+			return;
+		}
+
+		const isFreshPage = ! isEditedPostDirty() && isEditedPostEmpty();
+		if ( isFreshPage ) {
 			setIsInserterOpened( {
 				tab: 'patterns',
 				category: 'core/starter-content',
 			} );
 		}
-	}, [ postId, shouldEnable, setIsInserterOpened ] );
+	}, [
+		postId,
```

Comment:
The `postId` ensures the effect re-runs when the user switches pages in the Site Editor or creates a new page via Command Palette.